src/groupeddataframe/splitapplycombine.jl

```diff
@@ -741,74 +775,45 @@ function _combine(f::AbstractVector{<:Pair}, gd::GroupedDataFrame)
         # This can speed up some aggregates that would not trigger this on their own
         @assert gd.idx !== nothing
     end
-    res = map(f) do p
-        if isagg(p)
-            incol = gd.parent[!, first(p)]
-            agg = check_aggregate(last(p))
-            outcol = agg(incol, gd)
-            return idx_agg, outcol
-        else
-            fun = do_f(last(p))
-            if p isa Pair{<:ColumnIndex}
-                incols = gd.parent[!, first(p)]
-            else
-                df = gd.parent[!, collect(first(p))]
-                incols = NamedTuple{Tuple(names(df))}(eachcol(df))
-            end
-            firstres = do_call(fun, gd.idx, gd.starts, gd.ends, gd, incols, 1)
-            idx, outcols, _ = _combine_with_first(wrap(firstres), fun, gd, incols)
-            return idx, outcols[1]
-        end
-    end
+    res = Vector{Any}(undef, length(f))
+    for (i, p) in enumerate(f)
+           if isagg(p)
+               incol = gd.parent[!, first(p)]
+               agg = check_aggregate(last(p))
+               outcol = agg(incol, gd)
+               res[i] = idx_agg, outcol
+           else
+               fun = do_f(last(p))
+               if p isa Pair{<:ColumnIndex}
+                   incols = gd.parent[!, first(p)]
+               else
+                   df = gd.parent[!, first(p)]
+                   incols = Tuple(eachcol(df))
+               end
+               firstres = do_call(fun, gd.idx, gd.starts, gd.ends, gd, incols, 1)
+               idx, outcols, _ = _combine_with_first(wrap(firstres), fun, gd, incols)
+               res[i] = idx, outcols[1]
+           end
+      end
     # TODO: avoid recomputing idx for each pair when !all(isagg, f)
     idx = res[1][1]
     outcols = map(x -> x[2], res)
     if !all(x -> length(x) == length(outcols[1]), outcols)
         throw(ArgumentError("all functions must return values of the same length"))
     end
-    nams = [f[i] isa Pair{<:ColumnIndex} ?
-                Symbol(names(gd.parent)[index(gd.parent)[first(f[i])]],
-                       '_', funname(last(f[i]))) :
-                Symbol('x', i)
-            for i in 1:length(f)]
-    valscat = DataFrame(collect(AbstractVector, outcols), nams, makeunique=true)
+    valscat = DataFrame(collect(AbstractVector, outcols))
```

Comment:
Maybe just return the vector?

Reply:
I have rewritten the `_combine` functions to avoid this step.